meteor/server/api/ingest/ingestCache.ts

```diff
@@ -9,22 +9,29 @@ import { RundownId } from '../../../lib/collections/Rundowns'
 import { SegmentId } from '../../../lib/collections/Segments'
 import { PartId } from '../../../lib/collections/Parts'
 
-export function loadCachedRundownData (rundownId: RundownId, rundownExternalId: string): IngestRundown {
+export function loadCachedRundownData (rundownId: RundownId, rundownExternalId: string): LocalIngestRundown {
 	const cacheEntries = IngestDataCache.find({ rundownId: rundownId }).fetch()
 
-	const baseEntry = cacheEntries.find(e => e.type === IngestCacheType.RUNDOWN)
-	if (!baseEntry) throw new Meteor.Error(404, `Rundown "${rundownId}", (${rundownExternalId}) has no cached ingest data`)
+	const cachedRundown = cacheEntries.find(e => e.type === IngestCacheType.RUNDOWN)
+	if (!cachedRundown) throw new Meteor.Error(404, `Rundown "${rundownId}", (${rundownExternalId}) has no cached ingest data`)
 
-	const ingestRundown = baseEntry.data as IngestRundown
+	const ingestRundown = cachedRundown.data as LocalIngestRundown
+	ingestRundown.modified = cachedRundown.modified
 
 	const segmentMap = _.groupBy(cacheEntries, e => e.segmentId)
 	_.each(segmentMap, objs => {
 		const segmentEntry = objs.find(e => e.type === IngestCacheType.SEGMENT)
 		if (segmentEntry) {
-			const ingestSegment = segmentEntry.data as IngestSegment
-			_.each(objs, e => {
-				if (e.type === IngestCacheType.PART) {
-					ingestSegment.parts.push(e.data as IngestSegment)
+
+			const ingestSegment = segmentEntry.data as LocalIngestSegment
+			ingestSegment.modified = segmentEntry.modified
```

Comment:
Shouldnt the various parts in this file use the makeNewIngest* functions instead of doing it manually?

Reply:
There is a difference:
The makeNewIngest**-functions (with emphasis on "New") always updates the `modified` property. In this case, I want to use it from the db to avoid unnecessary data-updates down the line.